extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts

```diff
@@ -0,0 +1,295 @@
+
+    /**
+     * appinsights channel extension instance.
+     */
+    let channelPluginExt: IPlugin;
+
+    /**
+     * timestamp used to track number of seconds since webpage was loaded
+     */
+    let startTime: number = +new Date();
+
+    /**
+     * the config for this plugin
+     */
+    let _extensionConfig: IDebugPluginConfig = DebugPlugin.getDefaultConfig();
+
+    dynamicProto(DebugPlugin, this, (_self, base) => {
+      _self.initialize = (config: IDebugPluginConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?: ITelemetryPluginChain) => {
+        if (!_self.isInitialized()) {
+          base.initialize(config, core, extensions, pluginChain);
+
+          // 1. grabs each extension involved in instrumentation
+          for (let i = 0; i < extensions.length; i++) {
+            if (extensions[i].identifier === 'ApplicationInsightsAnalytics') {
+              analyticsExt = extensions[i];
+            }
+            else if (extensions[i].identifier === 'AjaxDependencyPlugin') {
+              ajaxDependencyExt = extensions[i];
+            }
+            else if (extensions[i].identifier === 'AppInsightsPropertiesPlugin') {
+              propertiesExt = extensions[i];
+            }
+            else if (extensions[i].identifier === 'AppInsightsChannelPlugin') {
+              channelPluginExt = extensions[i];
+            }
+          }
+
+          // 2. initialize trackers if they don't exist
+          const defaultConfig = DebugPlugin.getDefaultConfig();
+          const ctx = _self._getTelCtx();
+          const identifier = _self.identifier;
+          for (const field in defaultConfig) {
+              _extensionConfig[field] = () => ctx.getConfig(identifier, field, defaultConfig[field]());
+          }
+
+          trackers = _extensionConfig.trackers();
+
+          // 3. setup debugBin
+          const debugBinContainer = document.createElement("div");
+          debugBinContainer.className = 'debug-bin-container';
+          debugBinParent = new DebugBinParent(debugBinContainer, [], 0);
+
+          // 4. grab all relevant functions that need to be instrumented and store them in arrays
+          const diagLog = _self.diagLog();
+
+          debugBins = {};
+          const propertiesProtoFns: string[] = [];
+          const analyticsProtoFns: string[] = [];
+          const ajaxProtoFns: string[] = [];
+          const channelProtoFns: string[] = [];
+          const diagLogProtoFns: string[] = [];
+          for (const [ext, protoFns] of [
+            [analyticsExt, analyticsProtoFns],
+            [propertiesExt, propertiesProtoFns],
+            [ajaxDependencyExt, ajaxProtoFns],
+            [channelPluginExt, channelProtoFns],
+            [diagLog, diagLogProtoFns]
+          ] as any[]) {
+            for (const key of CoreUtils.objKeys(ext['__proto__'])) {
+              if (key.substring(0, 1) === '_') { continue; }
+              if (CoreUtils.isTypeof(ext[key], 'function')) {
```

Comment:
We already have a CoreUtils.isFunction(), please use that so we avoid having the "function" string from getting included (this is for better code minification support).